eventlet/green/thread.py

```diff
@@ -15,6 +15,8 @@
 LockType = Lock
 __threadcount = 0
 
+if sys.version_info > (3, 12):
```

Comment:
Since some versions of 3.12 have this, maybe use a `hasattr()`? Alternatively, maybe catchall logic that just re-exports everything in the `__thread` module, just in case?

Reply:
Done.

You are right the `hasattr` approach is better.

The catchall logic could be implemented in a follow patch and then we would be able to remove this condition.
However, I won't spend time on it for now, so the current fix could be considered as a short term solution to be landed with the next eventlet release. We will have time early in 2024 to implement a more robust logic.

Thanks for your feedback.